assets/dags/mwaa_dr/framework/model/base_table.py

```diff
@@ -347,19 +347,34 @@ def restore(self, **context):
         """
         backup_file = self.read(context)
 
-        restore_sql = ""
         if self.columns:
             restore_sql = f"COPY {self.name} ({', '.join(self.columns)}) FROM STDIN WITH (FORMAT CSV, HEADER FALSE, DELIMITER '|')"
         else:
             restore_sql = f"COPY {self.name} FROM STDIN WITH (FORMAT CSV, HEADER FALSE, DELIMITER '|')"
         print(f"Restore SQL: {restore_sql}")
 
         conn = settings.engine.raw_connection()
+        cursor = None
         try:
             cursor = conn.cursor()
-            cursor.copy_expert(restore_sql, backup_file)
-            conn.commit()
+            insert_counter = 0
+            with backup_file as file:
+                batch = []
+                for line in file:
```

Comment:
I believe multiple commits are necessary to commit smaller batches of data instead of one large operation. Please feel free to add your code suggestions, but I will not be able to test them as I will be going on holiday soon. I am happy to give ownership of this pull request to you if you'd like. I am working on getting the code coverage back to 100%, but I plan to move on to another topic afterward.

Reply:
<img width="577" alt="image" src="https://github.com/user-attachments/assets/6af3b924-51e0-4c50-aa12-61b919eb82c3">

Are you happy to add your proposal and test the code?

Reply:
Yes, I have implemented your suggestion to use `itertools.islice`. Thank you for the recommendation.